plugins/authentication/src/OAuthModel/model.tsx

```diff
@@ -0,0 +1,475 @@
+      get internetAccountType() {
+        return 'OAuthInternetAccount'
+      },
+      handlesLocation(location: UriLocation): boolean {
+        const validDomains = self.accountConfig.validDomains || []
+        return validDomains.some((domain: string) =>
+          location?.uri.includes(domain),
+        )
+      },
+      get generateAuthInfo() {
+        return {
+          internetAccountType: this.internetAccountType,
+          authInfo: {
+            authHeader: self.authHeader,
+            tokenType: this.tokenType,
+            configuration: self.accountConfig,
+          },
+        }
+      },
+    }))
+    .actions(self => ({
+      setCodeVerifierPKCE(codeVerifier: string) {
+        self.codeVerifierPKCE = codeVerifier
+      },
+      setErrorMessage(message: string) {
+        self.errorMessage = message
+      },
+      async fetchFile(locationUri: string, accessToken: string) {
+        if (!locationUri || !accessToken) {
+          return
+        }
+        return locationUri
+      },
+    }))
+    .actions(self => {
+      let resolve: Function = () => {}
+      let reject: Function = () => {}
+      let openLocationPromise: Promise<string> | undefined = undefined
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      let preAuthInfo: any = {}
+      return {
+        // opens external OAuth flow, popup for web and new browser window for desktop
+        async useEndpointForAuthorization() {
+          const config = self.accountConfig
+          const data: OAuthData = {
+            client_id: config.clientId,
+            redirect_uri: 'http://localhost:3000',
+            response_type: config.responseType || 'code',
+          }
+
+          if (config.scopes) {
+            data.scope = config.scopes
+          }
+
+          if (config.needsPKCE) {
+            const base64Encode = (buf: Buffer) => {
+              return buf
+                .toString('base64')
+                .replace(/\+/g, '-')
+                .replace(/\//g, '_')
+                .replace(/=/g, '')
+            }
+            const codeVerifier = base64Encode(crypto.randomBytes(32))
+            const sha256 = (str: string) => {
+              return crypto.createHash('sha256').update(str).digest()
+            }
+            const codeChallenge = base64Encode(sha256(codeVerifier))
+            data.code_challenge = codeChallenge
+            data.code_challenge_method = 'S256'
+
+            self.setCodeVerifierPKCE(codeVerifier)
+          }
+
+          if (config.hasRefreshToken) {
+            data.token_access_type = 'offline'
+          }
+
+          const params = Object.entries(data)
+            .map(([key, val]) => `${key}=${encodeURIComponent(val)}`)
+            .join('&')
+
+          const url = `${config.authEndpoint}?${params}`
+
+          if (isElectron) {
+            const model = self
+            const electron = require('electron')
+            const { BrowserWindow } = electron.remote
```

Comment:
`electron.remote` has been removed from the current version of Electron, so we can't use it. `window.open` might work, but isn't optimal since it opens in your existing browser window as a tab. What will probably need to happen is that we use `ipcRenderer.invoke()` and send a message to the Electron main process, and the main process will open a new window.